### PR TITLE
docs: add installation steps in README

### PR DESCRIPTION
**Pull request by felangel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,54 @@
 # Shorebird (un)installer
+
+**❗️ Note: This project is under heavy development. Things will change frequently and none of the code is ready for production use. We will do our best to keep the documentation up-to-date.**
+
+## Getting Started 🚀
+
+1. 🔑 Request an API Key
```
Comment (felangel) [inline README.md]: @eseidel should we be more specific here? Lmk what you think.

File: README.md [hunk 2]
```diff
@@ -1 +1,54 @@
...
+1. 🐦 Install Shorebird
+
+   ```
+   dart pub global activate --source git https://github.com/shorebirdtech/shorebird --git-path packages/shorebird_cli
```
Comment (eseidel) [inline README.md:11]: This assumes you have Dart installed. Which is valid today, but won't be. 

I had presumed this wolud instead just be a `shorbird` executable in bin/shorebird?

Comment (felangel) [inline README.md:11]: This is temporary until we have a proper `install.sh` and assumes users have Dart installed.

File: README.md [hunk 3]
```diff
@@ -1 +1,54 @@
...
+   **❗️Note**: If it's the first time, `shorebird` will download and build the shorebird engine which may take some time. The shorebird engine will be cached for subsequent runs.
+
+1. ☁️ Publish a new release
+
+   ```bash
+   # Publish the default release
+   shorebird publish
+
+   # Publish a specific release
+   shorebird publish <path/to/libapp.so>
```
Comment (eseidel) [inline README.md]: I think this mode is going away.

Comment (felangel) [reply README.md]: Sounds good, I'll remove it from the docs 😄

File: README.md (hunk 1 above)
Comment (eseidel) [reply README.md]: We don't have a process yet.  We could just say (Process TBD) or something. 🤷

File: README.md (hunk 2 above)
Comment (felangel) [reply README.md:11]: https://github.com/shorebirdtech/install/pull/1#discussion_r1128534758

Comment (eseidel) [reply README.md:11]: lgtm

**Pull request merged**